pyomo/contrib/gdpopt/config_options.py

```diff
@@ -75,136 +96,228 @@ def _add_OA_configs(CONFIG):
+        description=""""
+        callback to specify custom routines to initialize the
+        (MI)NLP subproblems.""",
+        doc="""
+        Callback to specify custom routines for initializing the (MI)NLP
+        subproblems. This method is called after the master problem solution
+        is fixed in the subproblem and before the subproblem is solved (or
+        pre-solved).
+
+        Accepts three arguments: the solver object, the subproblem GDPopt
+        utility block and the master problem GDPopt utility block. The master
+        problem contains the most recent master problem solution.
+
+        The return of this method will be unused: The method should directly
+        set the value of the variables on the subproblem
+        """
     ))
     CONFIG.declare("call_before_subproblem_solve", ConfigValue(
         default=_DoNothing,
-        description="callback hook before calling the subproblem solver"
+        description="callback hook before calling the subproblem solver",
+        doc="""
+        Callback called right before the (MI)NLP subproblem is solved.
+        Takes three arguments: The solver object, the subproblem and the
+        GDPopt utility block on the subproblem.
+
+        Note that unless you are *very* confident in what you are doing, the
+        subproblem should not be modified in this callback: it should be used
+        to interrogate the problem only.
+
+        To initialize the problem before it is solved, please specify a method
+        in the 'subproblem_initialization_method' argument.
+        """
     ))
     CONFIG.declare("call_after_subproblem_solve", ConfigValue(
         default=_DoNothing,
-        description="callback hook after a solution of the "
-                    "nonlinear subproblem"
+        description="""
+        callback hook after a solution of the
+        "nonlinear subproblem""",
+        doc="""
+        Callback called right after the (MI)NLP subproblem is solved.
+        Takes three arguments: The solver object, the subproblem, and the
+        GDPopt utility block on the subproblem.
+
+        Note that unless you are *very* confident in what you are doing, the
+        subproblem should not be modified in this callback: it should be used
+        to interrogate the problem only.
+        """
     ))
     CONFIG.declare("call_after_subproblem_feasible", ConfigValue(
         default=_DoNothing,
-        description="callback hook after feasible solution of "
-                    "the nonlinear subproblem"
-    ))
-    CONFIG.declare("algorithm_stall_after", ConfigValue(
-        default=2,
-        description="number of non-improving master iterations after which "
-                    "the algorithm will stall and exit."
+        description="""
+        callback hook after feasible solution of
+        the nonlinear subproblem""",
+        doc="""
+        Callback called right after the (MI)NLP subproblem is solved,
```

Comment:
Do we like acronyms in the option descriptions? iF not, edit NLP, MILP, MINLP, ... in other options.

Reply:
I'm neutral. I kind of like it on the assumption that anyone using advanced configurations in GDPopt will probably know what we mean, and the paren thing actually makes the algorithm choices clearer. But if you think writing things out is better, I'm happy to.

Reply:
I'm neutral as well. It just seemed a little mixed-up. Feel free to ignore this comment.